pkg/agent/config/node_config.go

```diff
@@ -268,12 +268,18 @@ func (nc *NetworkConfig) CalculateMTUDeduction(isIPv6 bool) int {
 	var mtuDeduction int
 	// When Multi-cluster Gateway is enabled, we need to reduce MTU for potential cross-cluster traffic.
 	if nc.TrafficEncapMode.SupportsEncap() || nc.EnableMulticlusterGW {
-		if nc.TunnelType == ovsconfig.VXLANTunnel {
-			mtuDeduction = vxlanOverhead
-		} else if nc.TunnelType == ovsconfig.GeneveTunnel {
-			mtuDeduction = geneveOverhead
-		} else if nc.TunnelType == ovsconfig.GRETunnel {
-			mtuDeduction = greOverhead
+		if nc.TrafficEncryptionMode == TrafficEncryptionModeWireGuard {
```

Comment:
Please help to add a unit test case in the `TestCalculateMTUDeduction`. Thanks.

Reply:
Removed these changes.